Lib/ufo2ft/featureWriters/markFeatureWriter.py

```diff
@@ -875,9 +876,39 @@ def _write(self):
             return False
 
         feaFile = self.context.feaFile
-        feaFile.statements.extend(newClassDefs)
-        # add empty line to separate classes from following statements
-        feaFile.statements.append(ast.Comment(""))
-        for _, feature in sorted(features.items()):
-            feaFile.statements.append(feature)
+
+        insertion_tag2index = ufo2ft.featureWriters.findFeatureInsertionMarkers(feaFile)
```

Comment:
this only supports "### INSERT ... ###" comments for the MarkFeatureWriter. What about KernFeatureWriter (or future feature writers that we haven't written yet)?
Shouldn't this parsing be done once before all automatic features are generated, instead of at the level of the singular feature writer?
Or maybe even better, instead of the current finding all insertion markers and building a mapping of tag:index, we could move this logic to a method of BaseFeatureWriter which gets called just before compilation from within BaseFeatureWriter.setContext method, and it will store the index of the insertion comment in the self.context namespace for that writer; so no need for a mapping, as the link between the statement index and the feature tag is implicit by the association with the current writer's context.

Reply:
actually, each feature writer can write multiple feature tags (e.g. "mark", "mkmk", or "kern" and "dist"), so we'd still need to store a mapping from tag to index in the writer's self.context. But each different writer should in theory only be interested in the insertion tags that pertains to it, not all of them. Unless... the placement of class definitions poses a problem (see the other comment https://github.com/googlefonts/ufo2ft/pull/352/files#r560061188)